File: app/src/main/java/tpcreative/co/qrscanner/common/MainSingleton.java
package tpcreative.co.qrscanner.common;

public class MainSingleton {
    private static MainSingleton instance ;
    private SingleTonMainListener listener;

    public static MainSingleton getInstance(){
        if (instance==null){
            synchronized (MainSingleton.class){
                if (instance==null){
                    instance = new MainSingleton();
                }
            }
        }
        return instance;
    }

    public void setListener(SingleTonMainListener listener){
        this.listener = listener;
    }

    public void isShowDeleteAction(boolean isDelete){
        if (listener!=null){
            listener.isShowDeleteAction(isDelete);
        }
    }

    public interface SingleTonMainListener{
        void isShowDeleteAction(boolean isDelete);
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/common/MainSingleton.kt
package tpcreative.co.qrscanner.common
class MainSingleton {
    private var listener: SingleTonMainListener? = null
    fun setListener(listener: SingleTonMainListener?) {
        this.listener = listener
    }

    fun isShowDeleteAction(isDelete: Boolean) {
        if (listener != null) {
            listener?.isShowDeleteAction(isDelete)
        }
    }

    interface SingleTonMainListener {
        fun isShowDeleteAction(isDelete: Boolean)
    }

    companion object {
        private var instance: MainSingleton? = null
        fun getInstance(): MainSingleton? {
            if (instance == null) {
                synchronized(MainSingleton::class.java) {
                    if (instance == null) {
                        instance = MainSingleton()
                    }
                }
            }
            return instance
        }
    }
}